addons/website_crm/controllers/main.py

```diff
@@ -60,6 +60,13 @@ def _handle_website_form(self, model_name, **kwargs):
     def insert_record(self, request, model, values, custom, meta=None):
         is_lead_model = model.model == 'crm.lead'
         if is_lead_model:
+            visitor = request.env['website.visitor']._get_visitor_from_request()
+            visitor_partner = visitor.partner_id
+            if visitor and visitor_partner:
+                fmt_partner_number = visitor_partner._phone_format(visitor_partner.phone) if visitor_partner.phone else False
+                # Here, 'phone' in values has already been formatted, see _handle_website_form.
+                if fmt_partner_number == values.get('phone', False) and visitor_partner.email == values['email_from']:
```

Comment:
We could add some pre-checks to skip some computation if values are not given. Idea would be: we need a phone in value (already formatted), and a normalized email (not sure if already formatted, anyway method does not cost anything). Would lead to something like:

```
# Here, 'phone' in values has already been formatted, see _handle_website_form
values_phone = values.get('phone') 
values_email_normalized = tools.email_normalize(values.get('email_from'))

# link to visitor partner only if email and phone values are set and match
# otherwise not enough information to decide
if values_phone and values_email_normalized:
    visitor = request.env['website.visitor']._get_visitor_from_request()
    visitor_partner = visitor.partner_id

    if visitor and visitor_partner:
        fmt_partner_number = ...
        if fmt_partner_number ==values_phone and visitor_partner.email_normalized = values_email_normalized:
            values['partner_id'] = visitor_partner.id
```

Reply:
Hello, I see your point. I agree we could skip a part of the computation but as I understand it, a partner may not have a phone? So we could do a condition on normalized email since, if I am not mistaken, a logged user should have an email, maybe something like : 
`           
            values_email_normalized = tools.email_normalize(values.get('email_from'))

            if values_email_normalized:
                visitor = request.env['website.visitor']._get_visitor_from_request()
                visitor_partner = visitor.partner_id

                if visitor and visitor_partner:
                    # Here, 'phone' in values has already been formatted, see _handle_website_form.
                    values_phone = values.get('phone', False)
                    fmt_partner_number = visitor_partner._phone_format(visitor_partner.phone) if visitor_partner.phone else False
                    
                    if fmt_partner_number == values_phone and visitor_partner.email_normalized == values_email_normalized:
                        values['partner_id'] = visitor_partner.id 
` 
Thanks !

Reply:
-> This way we can still write partner id on crm lead if partner has no phone number and do not fill it on the form?

Reply:
Oh I see ... globally: email is required and should match, phone is "a plus".

If I summarize:

  * email is required and should match partner email (if no given or no email on partner -> skip)
  * if phone in input
    * if phone on partner: must match
    * if no phone on partner: ok (additional information)
  * if no phone in input: ok (not giving it does not allow any check)

What do you think ?

Reply:
That sounds ok for me, if the field in not in values, it should not interact with the model indeed.
The only point that could be changed during testing is the additional information, but it makes sense from contact form to write phone on partner. That would become something like:

```
            values_email_normalized = tools.email_normalize(values.get('email_from'))
            if values_email_normalized:
                visitor = request.env['website.visitor']._get_visitor_from_request()
                visitor_partner = visitor.partner_id

                if visitor and visitor_partner and visitor_partner.email_normalized == values_email_normalized:
                    # Here, 'phone' in values has already been formatted, see _handle_website_form.
                    values_phone = values.get('phone')

                    if values_phone and visitor_partner.phone:
                        if visitor_partner._phone_format(visitor_partner.phone) == values_phone:
                            values['partner_id'] = visitor_partner.id
                    else:
                        values['partner_id'] = visitor_partner.id
```